### PR TITLE
chore(docs): add concepts doc, add filepath docs

### PR DESCRIPTION
**Pull request by toddbaert**
This PR:

- adds a new `concepts.md`, which includes the old (and small) `architecture.md`.
- adds more details docs and explanations about the various `syncProviders`.

### CODE REVIEW

File: docs/concepts.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+> "the total delay from the moment when the ConfigMap is updated to the moment when new keys are projected to the Pod can be as long as the kubelet sync period + cache propagation delay"
+
+Consider your individual requirements and select the configuration most appropriate for your needs. Note that the sync provider configuration to use can be configured globally and overridden per `featureflagconfiguration`. For details, see [the syncProvider documentation](./feature_flag_configuration.md#syncprovider).
```
Comment (beeme1mr) [inline docs/concepts.md:23]:
```suggestion
Consider your individual requirements and select the configuration most appropriate for your needs. Note that the sync provider configuration to use can be configured globally and overridden per `featureflagconfiguration`. For details, see [the sync provider documentation](./feature_flag_configuration.md#syncprovider).
```

File: docs/feature_flag_configuration.md [hunk 2]
```diff
@@ -6,24 +6,51 @@ The `FeatureFlagConfiguration` version `v1alpha2` CRD defines the a CR with the
...
-          value: "8080"
-    featureFlagSpec: 
-        flags: 
-            foo: 
-                state: "ENABLED"
-                variants: 
-                    bar: "BAR"
-                    baz: "BAZ"
-                defaultVariant: "bar"
+  flagDSpec:
```
Comment (beeme1mr) [inline docs/feature_flag_configuration.md:11]: I think `flagDSpec` is deprecated now. We should remove this example from the docs.

File: docs/feature_flag_configuration.md [hunk 3]
```diff
@@ -6,24 +6,51 @@ The `FeatureFlagConfiguration` version `v1alpha2` CRD defines the a CR with the
...
+      foo:
+        state: "ENABLED"
+        variants:
+          bar: "BAR"
+          baz: "BAZ"
+        defaultVariant: "bar"
 ```
 
-Within the CRD there are 2 main objects, namely the `flagDSpec` and the `featureFlagSpec`, both offering a different set of configuration for the injected `flagd` sidecars.
+Within the CRD there are 3 main objects, namely the `flagDSpec`, the `featureFlagSpec`, and the `syncProvider`, each offering a different set of configurations for the injected `flagd` sidecars.
```
Comment (beeme1mr) [inline docs/feature_flag_configuration.md:25]: Update this section is you end up remove `flagDSpec` from the example above.

File: docs/getting_started.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # Getting Started
 
-Once you have [installed the operator](./installation.md) you can follow this guide to deploy an example application demonstrating the operator.
+Once you have [understood the basic concepts](./concepts.md) and [installed the operator](./installation.md) you can follow this guide to deploy an example application demonstrating the operator.
```
Comment (beeme1mr) [inline docs/getting_started.md:3]: This sentence seems a bit wordy. Could you try and rephase it?

Comment (toddbaert) [reply docs/getting_started.md:3]: sure

File: docs/feature_flag_configuration.md (hunk 3 above)
Comment (toddbaert) [reply docs/feature_flag_configuration.md:25]: ok

File: docs/feature_flag_configuration.md (hunk 2 above)
Comment (toddbaert) [reply docs/feature_flag_configuration.md:11]: ok

File: docs/feature_flag_configuration.md (hunk 3 above)
Comment (toddbaert) [reply docs/feature_flag_configuration.md:25]: ok

File: docs/getting_started.md (hunk 4 above)
Comment (toddbaert) [reply docs/getting_started.md:3]: done

**Comment by toddbaert** [comment]
@beeme1mr please re-review.

**Review by beeme1mr — APPROVED** [review]

File: docs/feature_flag_configuration.md [hunk 5]
```diff
@@ -6,24 +6,43 @@ The `FeatureFlagConfiguration` version `v1alpha2` CRD defines the a CR with the
...
+      foo:
+        state: "ENABLED"
+        variants:
+          bar: "BAR"
+          baz: "BAZ"
+        defaultVariant: "bar"
 ```
 
-Within the CRD there are 2 main objects, namely the `flagDSpec` and the `featureFlagSpec`, both offering a different set of configuration for the injected `flagd` sidecars.
+Within the CRD there are 2 main objects, namely the `featureFlagSpec` and the `syncProvider` each offering a different set of configurations for the injected `flagd` sidecars.
```
Comment (skyerus) [inline docs/feature_flag_configuration.md:21]: Is it worth including the `syncProvider` in the above config?

File: docs/feature_flag_configuration.md [hunk 6]
```diff
@@ -6,24 +6,43 @@ The `FeatureFlagConfiguration` version `v1alpha2` CRD defines the a CR with the
...
 
 The `featureFlagSpec` is an object representing the flag configurations themselves, the documentation for this object can be found [here](https://github.com/open-feature/flagd/blob/main/docs/configuration/flag_configuration.md).
 
-The `flagDSpec` has one property called `envs` which contains a list of environment variables used to override the default start command values in `flagd`, the documentation for this configuration can be found [here](https://github.com/open-feature/flagd/blob/main/docs/configuration/configuration.md). These environment variables are also made available in the workload `Pod` for simplified configuration.
\ No newline at end of file
+## syncProvider
+
+The `syncProvider` specifies how the flag configuration will be supplied to flagd. It contains 2 optional members, `name` and `httpSyncConfiguration`.
+
+The default `syncProvider` is `kubernetes`, but can be configured globally at installation time via by modifying the `defaultSyncProvider` parameter or by setting the `sidecarConfiguration.defaultSyncProvider` helm value (`helm upgrade -i ofo openfeature/open-feature-operator  --set  sidecarConfiguration.defaultSyncProvider=filepath`).
```
Comment (skyerus) [inline docs/feature_flag_configuration.md:31]:
```suggestion
The default `syncProvider` is `kubernetes`, but can be configured globally at installation time by modifying the `defaultSyncProvider` parameter or by setting the `sidecarConfiguration.defaultSyncProvider` helm value (`helm upgrade -i ofo openfeature/open-feature-operator  --set  sidecarConfiguration.defaultSyncProvider=filepath`).
```

File: docs/feature_flag_configuration.md [hunk 7]
```diff
@@ -6,24 +6,43 @@ The `FeatureFlagConfiguration` version `v1alpha2` CRD defines the a CR with the
...
+
+### name
+
+| Value                  | Explanation                                                                                                                                                                                                                                                                                                |
+| ---------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| "kubernetes" (default) | Instruct flagd to query the Kubernetes API directly for the flag configuration custom resource(s) specified in the workload annotation (`openfeature.dev/featureflagconfiguration`). This configuration requires the flag sidecar (and therefor the workload pod) to be able to access the Kubernetes API. |
```
Comment (skyerus) [inline docs/feature_flag_configuration.md:37]:
```suggestion
| "kubernetes" (default) | Instruct flagd to query the Kubernetes API directly for the flag configuration custom resource(s) specified in the workload annotation (`openfeature.dev/featureflagconfiguration`). This configuration requires the flag sidecar (and therefore the workload pod) to be able to access the Kubernetes API. |
```

File: docs/feature_flag_configuration.md [hunk 8]
```diff
@@ -6,24 +6,43 @@ The `FeatureFlagConfiguration` version `v1alpha2` CRD defines the a CR with the
...
+| "filepath"             | Mounts the flag configuration custom resource(s) specified in the workload annotation (`openfeature.dev/featureflagconfiguration`) as volume mounted ConfigMaps, and configures flagd to watch them.                                                                                                       |
+| "http"                 | Retrives the flag configuration from the specified HTTP endpoint. Open Feature Operator does not automatically provide the specified configuration at this URL. Requires [`httpSyncConfiguration`](#httpSyncConfiguration) to be configured.                                                               |
```
Comment (skyerus) [inline docs/feature_flag_configuration.md:39]:
```suggestion
| "http"                 | Retrieves the flag configuration from the specified HTTP endpoint. Open Feature Operator does not automatically provide the specified configuration at this URL. Requires [`httpSyncConfiguration`](#httpSyncConfiguration) to be configured.                                                               |
```

File: docs/feature_flag_configuration.md (hunk 7 above)
Comment (toddbaert) [reply docs/feature_flag_configuration.md:37]: https://www.grammarly.com/blog/therefore-vs-therefor/#:~:text=Therefore%20is%20an%20adverb%20that,%E2%80%9D%20or%20%E2%80%9Cfor%20it.%E2%80%9D 

Dammit, you're right.

File: docs/feature_flag_configuration.md (hunk 5 above)
Comment (toddbaert) [reply docs/feature_flag_configuration.md:21]: Yes, will add.

File: docs/feature_flag_configuration.md [hunk 9]
```diff
@@ -6,24 +6,43 @@ The `FeatureFlagConfiguration` version `v1alpha2` CRD defines the a CR with the
...
-          value: "8080"
-    featureFlagSpec: 
-        flags: 
-            foo: 
-                state: "ENABLED"
-                variants: 
-                    bar: "BAR"
-                    baz: "BAZ"
-                defaultVariant: "bar"
+  featureFlagSpec:
```
Comment (toddbaert) [inline docs/feature_flag_configuration.md:11]:
```suggestion
  syncProvider:
    name: filepath # kubernetes, filepath or http 
  featureFlagSpec:
```

File: docs/feature_flag_configuration.md [hunk 10]
```diff
@@ -6,24 +6,45 @@ The `FeatureFlagConfiguration` version `v1alpha2` CRD defines the a CR with the
...
-    featureFlagSpec: 
-        flags: 
-            foo: 
-                state: "ENABLED"
-                variants: 
-                    bar: "BAR"
-                    baz: "BAZ"
-                defaultVariant: "bar"
+  syncProvider:
+    name: filepath # kubernetes, filepath or http
```
Comment (toddbaert) [inline docs/feature_flag_configuration.md:12]:
```suggestion
    name: filepath # kubernetes (default), filepath or http 
```

**Comment by toddbaert** [comment]
@skyerus please re-review when you have a sec.

**Review by skyerus — APPROVED** [review]
